x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/step_select_hosts.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo } from 'react';
+import type { EuiTabbedContentTab } from '@elastic/eui';
+import { EuiTabbedContent } from '@elastic/eui';
+import styled from 'styled-components';
+
+import { useGetAgentPolicies } from '../../../hooks';
+import type { AgentPolicy, NewAgentPolicy, PackageInfo } from '../../../types';
+import { AgentPolicyIntegrationForm } from '../components';
+import type { ValidationResults } from '../components/agent_policy_validation';
+
+import { incrementPolicyName } from '../../../services';
+
+import { StepSelectAgentPolicy } from './step_select_agent_policy';
+
+export enum SelectedPolicyTab {
+  NEW = 'new',
+  EXISTING = 'existing',
+}
+
+const StyledEuiTabbedContent = styled(EuiTabbedContent)`
+  [role='tabpanel'] {
+    padding-top: ${(props) => props.theme.eui.paddingSizes.m};
+  }
+`;
+
+interface Props {
+  agentPolicy: AgentPolicy | undefined;
+  updateAgentPolicy: (u: AgentPolicy | undefined) => void;
+  newAgentPolicy: Partial<NewAgentPolicy>;
+  updateNewAgentPolicy: (u: Partial<NewAgentPolicy>) => void;
+  withSysMonitoring: boolean;
+  updateSysMonitoring: (newValue: boolean) => void;
+  validation: ValidationResults;
+  packageInfo?: PackageInfo;
+  setHasAgentPolicyError: (hasError: boolean) => void;
+  updateSelectedTab: (tab: SelectedPolicyTab) => void;
+  selectedAgentPolicyId?: string;
+}
+
+export const StepSelectHosts: React.FunctionComponent<Props> = ({
+  agentPolicy,
+  updateAgentPolicy,
+  newAgentPolicy,
+  updateNewAgentPolicy,
+  withSysMonitoring,
+  updateSysMonitoring,
+  validation,
+  packageInfo,
+  setHasAgentPolicyError,
+  updateSelectedTab,
+  selectedAgentPolicyId,
+}) => {
+  let agentPolicies: AgentPolicy[] = [];
+  const { data: agentPoliciesData, error: err } = useGetAgentPolicies({
+    page: 1,
+    perPage: 1000,
+    sortField: 'name',
+    sortOrder: 'asc',
+    full: true,
+  });
+  if (err) {
+    // eslint-disable-next-line no-console
+    console.debug('Could not retrieve agent policies');
```

Comment:
could you clarify what do you mean by this?

Reply:
It's a nice to have, but we could report this error to our internal APM agent. This agent is enabled by default but we use it in some internal clusters. I'm thinking this would work:

```
import apm from '@elastic/apm-rum';
apm.captureError(err);
```

Reply:
I am not sure we should log|capture the error in a render, this could be call too many times, if we want to debug that|capture it should probably be in an `useEffect` block.

Reply:
I'm not too comfortable with introducing usage of apm here, it looks like hasn't been used before in fleet. Maybe we can create a new issue for this?